.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,340 @@
+# Contributing Guidelines for TerseTS
+
+TerseTS is a high-performance compression library written in [Zig](https://ziglang.org/) with bindings for other languages through a C interface. It uses the Apache‑2.0 license and includes a pre‑commit hook for building, formatting, and running tests. We welcome contributions of all kinds: bug reports, feature requests, code, documentation, or tests. The following guidelines summarize how to contribute effectively.
+
+## 1. Environment Setup
+
+We recommend consulting the relevant README and source file for language-specific contribution setup.
```

Comment:
Maybe just write "Consult the relevant README...." as "we recommend" provides no additional information.